alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteMonitorAction.kt

```diff
@@ -123,6 +126,34 @@ class TransportDeleteMonitorAction @Inject constructor(
             }
         }
 
+        /**
+         * Checks if the monitor is part of the workflow
+         *
+         * @param monitorId id of monitor that is checked if it is a workflow delegate
+         */
+        private suspend fun monitorIsNotInWorkflows(monitorId: String): Boolean {
+            val queryBuilder = QueryBuilders.nestedQuery(
+                Workflow.WORKFLOW_DELEGATE_PATH,
+                QueryBuilders.boolQuery().must(
+                    QueryBuilders.matchQuery(
+                        Workflow.WORKFLOW_MONITOR_PATH,
+                        monitorId
+                    )
+                ),
+                ScoreMode.None
+            )
+
+            val searchRequest = SearchRequest()
+                .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
+                .source(SearchSourceBuilder().query(queryBuilder).fetchSource(true))
+
+            val searchResponse: SearchResponse = client.suspendUntil { search(searchRequest, it) }
```

Comment:
We need to stash the context since the user would not have permissions to search the alerting config index directly.
reference: https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetMonitorAction.kt#L66-L72